### PR TITLE
rpc api: eth_getNatSpec

### PR DESCRIPTION
**Pull request by zelig**
* xeth, rpc: implement eth_getNatSpec for tx confirmations
* rename silly docserver -> httpclient
* eth/backend: httpclient now accessible via eth.Ethereum init-d via config.DocRoot
* cmd: introduce separate CLI flag for DocRoot (defaults to homedir)
* common/path: delete unused assetpath func, separate HomeDir func

fixes https://github.com/ethereum/go-ethereum/issues/1913

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Wed Oct 21 12:23:42 UTC 2015_

**Comment by tgerring** [comment]
Will this conflict with #1912 ?

**Comment by bas-vk** [comment]
@tgerring, no. The migration to the new rpc layer is work in progress.
@zelig, will this method also be implemented in the other nodes?

File: common/httpclient/httpclient.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		}
+	}
+	return false
+}
+
+func (self *HTTPClient) GetAuthContent(uri string, hash common.Hash) (content []byte, err error) {
+	// retrieve content
+	content, err = self.Get(uri, "")
+	if err != nil {
+		return
```
Comment (Gustav-Simonsson) [inline common/httpclient/httpclient.go]: Would recommend to avoid naked returns; afaik we're moving away from them in the code base unless there is some specific reason. @karalabe recently explained some reasons for when they could be used in a comment to another PR: https://github.com/ethereum/go-ethereum/pull/1917#discussion_r42972450

File: common/httpclient/httpclient.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+}
+
+// Get(uri, path) downloads the document at uri, if path is non-empty it
+// is interpreted as a filepath to which the contents are saved
+func (self *HTTPClient) Get(uri, path string) (content []byte, err error) {
+	// retrieve content
+	resp, err := self.Client().Get(uri)
+
+	defer func() {
```
Comment (Gustav-Simonsson) [inline common/httpclient/httpclient.go]: Can move this after the error check, since if `Get` returns an error `resp` does not need to be closed.

File: common/httpclient/httpclient.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		if resp != nil {
+			resp.Body.Close()
+		}
+	}()
+	if err != nil {
```
Comment (Gustav-Simonsson) [inline common/httpclient/httpclient.go]: Would also avoid naked returns here.

File: common/httpclient/httpclient.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		return
+	}
+	content, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	if path != "" {
+		var abspath string
+		abspath, err = filepath.Abs(path)
```
Comment (Gustav-Simonsson) [inline common/httpclient/httpclient.go]: Short-circuit on `err` returned from `Abs` and also check err returned from `WriteFile`

File: common/httpclient/httpclient.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+		ioutil.WriteFile(abspath, content, 0700)
```
Comment (Gustav-Simonsson) [inline common/httpclient/httpclient.go]: Do written files need execute? If not please change to 0600.

File: rpc/api/admin.go [hunk 6]
```diff
@@ -25,7 +25,7 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/compiler"
-	"github.com/ethereum/go-ethereum/common/docserver"
+	// "github.com/ethereum/go-ethereum/common/httpclient"
```
Comment (Gustav-Simonsson) [inline rpc/api/admin.go]: can be removed?

**Comment by bas-vk** [comment]
:+1:

**Comment by frozeman** [comment]
Could you please give an example in and put output for this new endpoint so i can judge?
Im currently busy with the custom coin branch so i don't want to switch now

**Comment by Gustav-Simonsson** [comment]
:+1:
